### ISSUE TITLE
Clear items from DataStore which are no longer included in selective sync expression

### ISSUE THREAD

**Issue by mlxyz** [aws-amplify/amplify-js#7134]
**Is your feature request related to a problem? Please describe.**
When applying a more restrictive selective sync expression for DataStore, the existing content of the local storage is retained (see https://docs.amplify.aws/lib/datastore/sync/q/platform/js#reevaluate-expressions-at-runtime). 
In our application, a portion of data items we sync with DataStore is newly generated every day and the sync expression only includes the items which are relevant for the current day. Without a way of clearing out "old" items which are no longer included in the sync expressions, the local storage of clients will grow indefinitely over time impacting performance and user experience.

**Describe the solution you'd like**
We would be fine with either of these solutions:
- DataStore automatically clears items from local storage which are no longer included in a sync expression. 
- Allow developers to manually remove items from the DataStore (by id or using a selective expression) without deleting them from DynamoDB.

**Describe alternatives you've considered**
We have considered calling `DataStore.clear()` to remove old content when changing the sync expressions (as mentioned in the documentation), but this is inefficient as DataStore then also has to resync data items which should be retained over longer time periods.

**Comment by iartemiev** [comment]
Hey @mlxyz, we're currently looking into ways of accomplishing this. Will keep this issue updated with any developments!

**Comment by mir1198yusuf** [comment]
@iartemiev I think `Datastore.stop > Datastore.clear > Datastore.start `would work as expected .

**Comment by mlxyz** [comment]
@mir1198yusuf This does work, but the problem is that DataStore resyncs **everything** - not just the sync expressions that have changed.

**Comment by mir1198yusuf** [comment]
yeah correct, so what you want in this issue is to run a query **locally** only to discard those items.

**Comment by mlxyz** [comment]
@mir1198yusuf Yes, that's basically it. Even a way to clear individual items from DataStore only by id would suffice.

**Comment by skontou** [comment]
I would prefer to selectively clear a subset of data similar to https://docs.amplify.aws/lib/datastore/sync/q/platform/ios#clear-local-data
This would cover the scenario of clearing user data as recommended https://docs.amplify.aws/lib/datastore/other-methods/q/platform/ios but allow you to keep static data

**Comment by Gab1988** [comment]
In my case I wanted to ensure that after all logins (if online) data is always fresh 
E.g. records update from external API on DynamoDB to be reflected on the local as it looks to me that if the update is not triggered by the client then there is no way for the client to know data has changed on the central
I'm pretty new with DataStore but it would be nice to allow to query only central under certain scenarios; happy to hear thoughts here :-)

Anyhow if you try to run the following on React Native Web at app initialisation it does not work

`useEffect(() => {here...},[
await DataStore.clear();
...do something
]}`

Digging into the code the following switch checks for the initialisation of the storage that the clear method does not invoke for you
`//datastore.js in node_modules`
<img width="851" alt="image" src="https://user-images.githubusercontent.com/47523111/126735678-6f878989-b67c-4a7e-87b1-9cabd8bd8b94.png">

**The solution is to explicitly call `DataStore.start()` before calling the clear method**

Keep in mind that this also deletes all local data including potentially user data

Let me know if you have found a better solution

**Comment by seanhamstra** [comment]
Any word on this? We have the same need but with amplify-flutter.

**Comment by thoschwa** [comment]
Any update on this?
